internal/clients/vault_secrets_preview.go

```diff
@@ -61,27 +61,41 @@ func OpenVaultSecretsAppSecrets(ctx context.Context, client *Client, loc *shared
 
 	var secrets *secret_service.OpenAppSecretsOK
 	var err error
+	var result []*secretmodels.Secrets20231128OpenSecret
+	nextPage := ""
 	for attempt := 0; attempt < retryCount; attempt++ {
-		secrets, err = client.VaultSecretsPreview.OpenAppSecrets(params, nil)
-		if err != nil {
-			var serviceErr *secret_service.OpenAppSecretDefault
-			ok := errors.As(err, &serviceErr)
-			if !ok {
+		for {
+			if nextPage != "" {
```

Comment:
is the if statement necessary? We can send an empty pagination token

Reply:
Yeah, removed the extra variable